extensions/amp-sidebar/1.0/component.js

```diff
@@ -166,11 +167,81 @@ function SidebarWithRef(
         >
           <div className={classes.backdropOverscrollBlocker}></div>
         </div>
-      </>
-    )
+      </div>
+    </>
   );
 }
 
 const Sidebar = forwardRef(SidebarWithRef);
 Sidebar.displayName = 'Sidebar'; // Make findable for tests.
 export {Sidebar};
+
+/**
+ * @param {!SidebarDef.SidebarNavToolbar} props
+ * @return {PreactDef.Renderable}
+ */
+export function SidebarNavToolbar({
+  toolbar,
+  'toolbar-target': toolbarTarget,
+  children,
+  ...rest
+}) {
+  const ref = useRef();
+  const [targetEl, setTargetEl] = useState();
+  const [, setReRender] = useState(true);
+
+  // Trigger re-render each time the window is resized
+  useEffect(() => {
+    const navElement = ref.current;
+    if (!navElement) {
+      return;
+    }
+    const document = navElement.ownerDocument;
+    if (!document) {
+      return;
+    }
+    const window = document.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const reRenderFunction = debounce(
+      window,
+      () => setReRender((prev) => !prev),
+      100
+    );
+    window.addEventListener('resize', reRenderFunction);
+    return () => {
+      window.removeEventListener('resize', reRenderFunction);
+    };
+  }, []);
+
+  useEffect(() => {
+    const navElement = ref.current;
+    if (!navElement) {
+      return;
+    }
+    const document = navElement.ownerDocument;
+    if (!document) {
+      return;
+    }
+    const window = document.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const matches = window.matchMedia(toolbar).matches;
+    const selector = `#${CSS.escape(toolbarTarget)}`;
+    const newTargetEl = document.querySelector(selector);
```

Comment:
What do you think about passing refs instead? I'm not sure how "normal" this is to do?

Reply:
I'm open to it.  Would you mind providing an example of what you mean?

Reply:
Assume `toolbarTarget` is type `HTMLElement` instead of a string id value that you have to query the document for. Then we also should not need this effect or `targetEl`/`setTargetEl` as it will be given as a prop.